fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1124,69 +1124,106 @@ ACTOR Future<bool> rebalanceTeams( DDQueueData* self, int priority, Reference<ID
 ACTOR Future<Void> BgDDMountainChopper( DDQueueData* self, int teamCollectionIndex ) {
 	state double checkDelay = SERVER_KNOBS->BG_DD_POLLING_INTERVAL;
 	state int resetCount = SERVER_KNOBS->DD_REBALANCE_RESET_AMOUNT;
+	state Transaction tr(self->cx);
 	loop {
-		wait( delay(checkDelay, TaskPriority::DataDistributionLaunch) );
-		if (self->priority_relocations[PRIORITY_REBALANCE_OVERUTILIZED_TEAM] < SERVER_KNOBS->DD_REBALANCE_PARALLELISM) {
-			state Optional<Reference<IDataDistributionTeam>> randomTeam = wait( brokenPromiseToNever( self->teamCollections[teamCollectionIndex].getTeam.getReply( GetTeamRequest( true, false, true ) ) ) );
-			if( randomTeam.present() ) {
-				if( randomTeam.get()->getMinFreeSpaceRatio() > SERVER_KNOBS->FREE_SPACE_RATIO_DD_CUTOFF ) {
-					state Optional<Reference<IDataDistributionTeam>> loadedTeam = wait( brokenPromiseToNever( self->teamCollections[teamCollectionIndex].getTeam.getReply( GetTeamRequest( true, true, false ) ) ) );
-					if( loadedTeam.present() ) {
-						bool moved = wait( rebalanceTeams( self, PRIORITY_REBALANCE_OVERUTILIZED_TEAM, loadedTeam.get(), randomTeam.get(), teamCollectionIndex == 0 ) );
-						if(moved) {
-							resetCount = 0;
-						} else {
-							resetCount++;
+		try {
+			tr.setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+			tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+			Optional<Value> val = wait(tr.get(rebalanceDDIgnoreKey));
+			if (val.present()) {
+				continue;
```

Comment:
I noted this in my another comment, but since it is outdated and the new code makes this more clear, I'll restate it. We should probably add a delay before the continue so that this doesn't lead the key in a tight loop.

Reply:
Good point.

Reply:
I am going to move the `wait(delay(checkDelay, TaskPriority::DataDistributionLaunch));` to be the first line of the try block.